### PR TITLE
Remove not longer required stubs

### PR DESCRIPTION
**Pull request by alexander-schranz**
This pull request replaces https://github.com/Jan0707/phpstan-prophecy/pull/298 which did replacer `@template` with `@template-covariant`

Since Prophecy 1.17.0 the template are correctly added to prophecy itself and so the stubs are not longer required. Thx to @stof.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -10,18 +10,18 @@
     }
   ],
   "require": {
-    "php": "^7.1 || ^8.0",
+    "php": "^7.2 || ^8.0",
```
Comment (alexander-schranz) [inline composer.json:13]: the stubs are added in 1.17.0 which only supports version from ^7.2 so we not longer can test on 7.1

**Comment by localheinz** [comment]
@alexander-schranz 

Would you be interested in #285?

**Comment by alexander-schranz** [comment]
@localheinz Sure it looks like it is not a lot of work todo here and I'm if some issue appear it looks like I'm already the first one stumble over it as I use prophecy with the plugin in most projects.

File: composer.json [hunk 2]
```diff
@@ -1,28 +1,38 @@
...
-    "phpunit/phpunit": "<6.0.0,>=10.0.0"
-  },
   "require-dev": {
     "ergebnis/composer-normalize": "^2.1.1",
     "ergebnis/license": "^1.0.0",
     "ergebnis/php-cs-fixer-config": "~2.2.0",
-    "phpspec/prophecy": "^1.7.0",
-    "phpunit/phpunit": "^6.0.0 || ^7.0.0 || ^8.0.0 || ^9.0.0"
+    "phpspec/prophecy": "^1.17.0",
+    "phpunit/phpunit": "^8.0.0 || ^9.0.0"
```
Comment (alexander-schranz) [inline composer.json:21]: prophecy 1.17 does only support PHPUnit 8, 9 (10 soon). So we can not longer test against 6 and 7.

**Review by stof — COMMENTED** [review]
I'm actually wondering whether this extension is still needed at all. Prophecy also has generic signatures for `prophesize()` methods

**Comment by jseparovic1** [comment]
> I'm actually wondering whether this extension is still needed at all. Prophecy also has generic signatures for `prophesize()` methods

I verified this and there is it still doesn't work without the extension (PHP 8.2, PhpUnit 10, phpstan 1.10). 

Test case:
```php
final class SimpleProphecyTestCase extends TestCase
{
    use ProphecyTrait;

    public function testCreatedViaTrait(): void
    {
        $foo = $this->prophesize(Foo::class);

        self::assertSame('foo', $foo->foo());
    }

     public function testCreatedInline(): void
    {
        $foo = (new Prophet())->prophesize(Foo::class);

        self::assertSame('foo', $foo->foo());
    }
}

class Foo
{
    public function foo(): string
    {
        return 'foo';
    }
}
```

Output:
```bash
 ------ -----------------------------------------------------------------------------------
  Line   SimpleProphecyTestCase.php
 ------ -----------------------------------------------------------------------------------
  20     Call to an undefined method Prophecy\Prophecy\ObjectProphecy::foo().
  27     Call to an undefined method Prophecy\Prophecy\ObjectProphecy<AppTest\Foo>::foo().
 ------ -----------------------------------------------------------------------------------
```

To make dynamic return types working `prophesize` method should return `T` instead of `ObjectProphecy<T>` but this won't be technically correct. 

Note that method in `ProphecyTrait` does not specify generic types 🤔 

It seems this extension is still required.

**Comment by InvisibleSmiley** [comment]
> I'm actually wondering whether this extension is still needed at all. Prophecy also has generic signatures for `prophesize()` methods

Two answers to this question.
1. When using the ProphecyTrait (`$this->prophesize`) instead of `(new Prophet())->prophesize`, the Prophecy generics are not used because the `prophesize` method override does not include them.
2. Even when the ProphecyTrait override PHPDoc is amended or the Prophecy method is used directly, without this extension here I get lots of PHPStan errors like `Call to an undefined method Prophecy\Prophecy\ObjectProphecy<SomeClass>::someMethod()` in cases where I create a prophecy object and then use like `$prophecy->someMethod()->willReturn('something');`

**Comment by stof** [comment]
ah indeed. I forgot the part covering the magic calls.

But I think WillExtendOrImplementDynamicReturnTypeExtension and ProphesizeDynamicReturnTypeExtension are not needed anymore.

> Note that method in `ProphecyTrait` does not specify generic types

this should be solved by doing a PR on the prophecy-phpunit repo IMO.

**Comment by InvisibleSmiley** [comment]
> > Note that method in `ProphecyTrait` does not specify generic types
> 
> this should be solved by doing a PR on the prophecy-phpunit repo IMO.

Obviously. :)

**Comment by stof** [comment]
And the TypeNodeResolverExtension should probably be deprecated because it is about transforming types like `@var ObjectProphecy|Foo` into `ObjectProphecy<Foo>` but there is no valid reason to write `ObjectProphecy|Foo` in phpdoc anymore as a hack (especially given that phpstan will complain about the missing generic type for `ObjectProphecy`)

**Comment by stof** [comment]
Actually, I just did the PR for the ProphecyTrait myself: https://github.com/phpspec/prophecy-phpunit/pull/63

**Comment by alexander-schranz** [comment]
close in favor of #348 where we removed the not longer required stubs

**Pull request closed without merging**